Add @tag documentation (#2033)
apollographql/federation: docs/source/federation-spec.md

  hidden: String @inaccessible
}
```

### `@tag`

```graphql

> directive @tag(from: String!) on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION

Reviewer: **blocker** This should be marked as `repeatable` and the input param is called `name`, to align with what's in code / the spec https://specs.apollo.dev/tag/v0.2/#@tag

```suggestion
directive @tag(name: String!) repeatable on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
```

```

> The `@tag` directive is used to provide a mechanism for applying arbitrary string metadata to the fields and types of a schema. Tags will be propagated up into composed supergraphs. This metadata is potentially useful throughout the schema’s lifecycle, including, but not limited to, processing, static analysis, and documentation.

Reviewer 2: ```suggestion
The `@tag` directive provides a mechanism for applying arbitrary string metadata to a schema's types and fields. This metadata can be useful throughout the schema’s lifecycle, including (but not limited to) processing, static analysis, and documentation.

Tags applied in subgraph schemas are preserved in the composed supergraph schema.
```

```graphql
extend schema

>     @link(url: "https://specs.apollo.dev/federation/v2.0", import: [""@tag"])

Reviewer 2: ```suggestion
    @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@tag"])
```

type Query {
  customer(id: String!): Customer @tag(name: "team-customers")
  employee(id: String!): Employee @tag(name: "team-admin")
}

interface User @tag(name: "team-accounts") {
  id: String!
  name: String!
}

type Customer implements User @tag(name: "team-customers") {
  id: String!
  name: String!

>   cart: [Product!] @tag(name: "team-shopping-cart")

Reviewer: **suggestion** `type Product` isn't defined in this schema. Just so we can make sure these examples compose, maybe it would be best to remove this field and put a tag on a different field.

```suggestion
```
